autonomi/src/client/data_types/scratchpad.rs

```diff
@@ -112,170 +126,209 @@ impl Client {
         Ok(pad)
     }
 
-    /// Returns the latest found version of the scratchpad for that secret key
-    /// If none is found, it creates a new one locally
-    /// Note that is does not upload that new scratchpad to the network, one would need to call [`Self::scratchpad_create`] to do so
-    /// Returns the scratchpad along with a boolean indicating if that scratchpad is new or not
-    pub async fn get_or_create_scratchpad(
-        &self,
-        public_key: &PublicKey,
-        content_type: u64,
-    ) -> Result<(Scratchpad, bool), PutError> {
-        let pad_res = self.scratchpad_get_from_public_key(public_key).await;
-        let mut is_new = true;
-
-        let scratch = if let Ok(existing_data) = pad_res {
-            info!("Scratchpad already exists, returning existing data");
-
-            info!(
-                "scratch already exists, is version {:?}",
-                existing_data.count()
-            );
-
-            is_new = false;
-
-            if existing_data.owner() != public_key {
-                return Err(PutError::ScratchpadBadOwner);
-            }
-
-            existing_data
-        } else {
-            trace!("new scratchpad creation");
-            Scratchpad::new(*public_key, content_type)
-        };
-
-        Ok((scratch, is_new))
-    }
-
-    /// Create a new scratchpad to the network
-    /// Returns the cost of the scratchpad and the address of the scratchpad
-    pub async fn scratchpad_create(
+    /// Manually store a scratchpad on the network
+    pub async fn scratchpad_put(
         &self,
         scratchpad: Scratchpad,
         payment_option: PaymentOption,
-    ) -> Result<(AttoTokens, ScratchpadAddress), PutError> {
-        let scratch_address = scratchpad.network_address();
-        let address = *scratchpad.address();
-        let scratch_key = scratch_address.to_record_key();
+    ) -> Result<(AttoTokens, ScratchpadAddress), ScratchpadError> {
+        let address = scratchpad.address();
 
         // pay for the scratchpad
-        let (receipt, _skipped_payments) = self
+        let xor_name = address.xorname();
+        let size = size_of::<Scratchpad>() + scratchpad.payload_size();
```

Comment:
better make this as a `utility function` within the protocol ScratchPad definition mod.
to avoid future mismatched usage, for example, other places take `scratchpad.bytes_for_sign().len()` as the `size of a scratchpad instance`

Reply:
Make sense